docs/css-style-guide.md

```diff
@@ -0,0 +1,92 @@
+Hypothesis CSS Style Guide
+==========================
+
+Hypothesis' CSS generally follows the principles of [Trello's CSS Style Guide](https://github.com/trello/trellisheets)
+for writing maintainable styling. In particular:
+
+ * Use simple, non-nested class selectors where possible
+ * Organize CSS into components. Unlike Trello, we use BEM-style naming (see below) and do not indent selectors
+ * Be careful with CSS preprocessor features. Unlike Trello, we use SASS but the principle
+   of using 'vanilla' CSS and being careful with preprocessor features still applies.
+   See the note about `@extends` below
+ * Use naming conventions to indicate which classes are purely presentational and which are referenced in code
+
+## File Structure
+
+The SASS for Hypothesis projects is organized into several parts:
+
+ * A 'root' file which imports all of the components used in the site, structured like this:
+
+```scss
+// Variables and mixins
+@import 'partials/base';
+
+// Reset default browser styles
+@import 'partials/reset';
+// Basic universal styling for elements (eg. links)
+@import 'partials/elements';
+
+// Components
+@import 'partials/input-field';
+@import 'partials/fancy-modal';
+@import 'partials/a-widget';
+```
+
+ * A 'base' partial which imports all variables and mixins used by other components
+ * A 'variables' partial which defines the colors, typography, grid spacing, animation durations and other foundational elements of the design system
+ * Component partials which define the styles for re-usable components, including the different variations of a component and media queries for responsive layout. These may assume
+   the existence of other variables or mixins but should not import anything themselves.
+
+## Selectors
+
+ * Use simple non-nested class selectors for the majority of styling
+ * Descendant selectors should be avoided. They [are slow](https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Writing_efficient_CSS#Avoid_the_descendant_selector.21) and more importantly, make it easy to create unintended results due to naming clashes or [specificity conflicts](http://css.maxdesign.com.au/selectutorial/advanced_conflict.htm).
```

Comment:
Yes. The reason I originally included both is because in SASS you can write a descendant selector either directly or via nesting. Looking at styling issues I've fixed in the past, I don't think the hazards of nesting (which visually looks nice in SASS) was obvious. Perhaps best to spell this out with an example?